boefjes/plugins/kat_webpage_analysis/main.py

```diff
@@ -45,21 +40,64 @@ def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
                 ]
             )
 
+    body_mimetypes = {"openkat-http/body"}
     try:
-        response = session.get(
-            uri,
-            headers={"Host": hostname, "Accept": "application/json"},
-            verify=False,
-            allow_redirects=False,
-        )
-        result = {
-            # "content": response.content,
-            "cookies": response.cookies.get_dict(),
-            "headers": dict(response.headers),
-            "code": response.status_code,
-            "text": response.text,
-        }
-    except requests.exceptions.ConnectionError:
-        result = {}
-
-    return [(set(), json.dumps(result))]
+        response = do_request(hostname, session, uri, useragent)
+    except requests.exceptions.RequestException as request_error:
+        return [({"openkat-http/error"}, str(request_error))]
+
+    if "content-type" in response.headers:
+        allowed_content_types = [
+            "application/ld+json",
+            "application/json",
+            "image/jpeg",
+            "image/jpg",
+            "image/gif",
+            "image/png",
+            "image/bpm",
+            "image/ico",
+            "text/html",
+            "text/plain",
+            "text/css",
+            "text/csv",
+            "text/javascript",
+            "text/xml",
+        ]
+        content_type = response.headers.get("content-type")
```

Comment:
Since you already know the key exists:
```suggestion
        content_type = response.headers["content-type"]
```